explorer/websocket.go

```diff
@@ -192,22 +189,13 @@ func (wsh *WebsocketHub) pingClients() chan<- struct{} {
 	return stopPing
 }
 
-func safeClose(cc hubSpoke) {
-	select {
-	case _, ok := <-cc:
-		if !ok {
-			log.Debug("Channel already closed!")
-			return
-		}
-	default:
-	}
-	close(cc)
-}
-
-// Stop kills the run() loop and unregisteres all clients (connections).
+// Stop kills the run() loop and unregisters all clients (connections).
 func (wsh *WebsocketHub) Stop() {
-	// end the run() loop, allowing in progress operations to complete
+	// End the run() loop, allowing in-progress operations to complete.
 	wsh.quitWSHandler <- struct{}{}
+	// Lastly close the hub relay channel sine the quitWSHandler signal is
+	// handled in the Run loop.
+	close(wsh.HubRelay)
```

Comment:
As in pubsub/websocket.go, close the `HubRelay` channel so that `(*WebsocketHub).run` will return.